File: app/src/main/java/com/example/monpetitterroir/model/Recipe.java
package com.example.monpetitterroir.model;

import java.util.List;

/**
 * Classe modélisant une recette de cuisine
 */
public class Recipe {
    String id;
    String image;
    String title;
    String likes;
    List<Ingredient> usedIngredients;

    public Recipe(String id, String image, String title, String likes, List<Ingredient> usedIngredients) {
        this.id = id;
        this.image = image;
        this.title = title;
        this.likes = likes;
        this.usedIngredients = usedIngredients;
    }

    public String getId() {
        return id;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getLikes() {
        return likes;
    }

    public void setLikes(String likes) {
        this.likes = likes;
    }

    public List<Ingredient> getUsedIngredients() {
        return usedIngredients;
    }

    public void setUsedIngredients(List<Ingredient> usedIngredients) {
        this.usedIngredients = usedIngredients;
    }
}

File: app/src/main/java/com/example/monpetitterroir/model/Recipe.kt
package com.example.monpetitterroir.model

import androidx.annotation.StringRes
import retrofit2.http.Url

/**
 * A recipe with name, image, likes, ingredients
 * @param id recipe unique id from the API
 * @param image recipe image url
 * @param likes recipe number of likes
 * @param usedIngredients ingredients from the search, used in the recipe
 * @param missedIngredients list of ingredients not in the search, used in the recipe
 */
data class Recipe(
 //       @StringRes val id: Int,
        @Url val image: String,
        val title: String,
  //      val likes: Int,
     //   val usedIngredients: List<Ingredient>,
    //    val missedIngredients : List<Ingredient>
) {
}
